### PR TITLE
Bjorncs/controller authorization

### PR DESCRIPTION
**Pull request by bjorncs**
Remove old authorization logic from controller. Include `ControllerAuthorizationFilter` as part of all `ContainerTester` based unit tests.

FYI @tokle

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java [hunk 1]
```diff
@@ -1046,26 +1025,31 @@ private Slime toSlime(InputStream jsonStream) {
...
+                .map(AthenzPrincipal::getIdentity)
+                .filter(AthenzUser.class::isInstance)
+                .map(AthenzUser.class::cast)
+                .map(AthenzUser::getName)
+                .map(UserId::new);
+    }
+
+    private static AthenzPrincipal getUserPrincipal(HttpRequest request) {
+        return Optional.ofNullable(request.getJDiscRequest().getUserPrincipal())
+                .map(AthenzPrincipal.class::cast)
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: Missing `isInstance` check before cast.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/ApplicationRequestToDiscFilterRequestWrapper.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Wraps an {@link Request} into a {@link DiscFilterRequest}. Only a few methods are supported.
+ * Changes are not propagated; updated request instance must be retrieved through {@link #getUpdatedRequest()}.
+ *
+ * @author bjorncs
+ */
+class ApplicationRequestToDiscFilterRequestWrapper extends DiscFilterRequest {
```
Comment (mpolden) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/ApplicationRequestToDiscFilterRequestWrapper.java]: Missing `public`.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/ContainerTester.java [hunk 3]
```diff
@@ -96,11 +103,31 @@ public void assertResponse(Supplier<Request> request, String expectedResponse, i
...
     public void assertResponse(Request request, String expectedResponse, int expectedStatusCode) throws IOException {
-        Response response = container.handleRequest(request);
+        FilterResult filterResult = invokeSecurityFilters(request);
+        request = filterResult.request;
+        Response response = filterResult.response != null ? filterResult.response : container.handleRequest(request);
         assertEquals(expectedResponse, response.getBodyAsString());
         assertEquals("Status code", expectedStatusCode, response.getStatus());
     }
 
+    // Hack to run request filters as part of the request processing chain.
```
Comment (mpolden) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/ContainerTester.java]: Nice, I was wondering why I couldn't get this to work in another test.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/ResponseHandlerToApplicationResponseWrapper.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * A {@link ResponseHandler} that caches the response content and
+ * converts a {@link Response} to {@link com.yahoo.application.container.handler.Response}.
+ *
+ * @author bjorncs
+ */
+class ResponseHandlerToApplicationResponseWrapper implements ResponseHandler {
```
Comment (mpolden) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/ResponseHandlerToApplicationResponseWrapper.java]: Missing `public`.

**Review by mpolden — COMMENTED** [review]
Nice cleanup!

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java (hunk 1 above)
Comment (bjorncs) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: It should fail hard if it's not an `AthenzPrincipal`. I agree though that `ClassCastException` is not the best way of doing that...

**Comment by bjorncs** [comment]
@mpolden PTAL

**Comment by mpolden** [comment]
LGTM. Merge when green.

**Pull request merged**